draft-ietf-core-coap-tcp-tls.md

```diff
@@ -1456,6 +1442,45 @@ Subprotocol Definition.
 
 --- back
 
+# Updates to RFC7641
+
+## Notifications and Reordering
+
+When using the Observe Option with CoAP over UDP, notifications from the server set the option
+value to an increasing sequence number for reordering detection on the client since messages
+can arrive in a different order than they were sent. This sequence number is not required
+for CoAP over reliable transports since the TCP protocol ensures reliable and ordered delivery
+of messages. The value of the Observe Option in 2.xx notifications MAY be empty on transmission
+and MUST be ignored on reception.
+
+## Transmission and Acknowledgements
+
+For CoAP over UDP, server notifications to the client can be confirmable or non-confirmable.
+A confirmable message requires the client to either respond with an acknowledgement message or
+a reset message. An acknowledgement message indicates that the client is alive and wishes to receive
+further notifications. A reset message indicates that the client does not recognize
+the token which causes the server to remove the associated entry from the list of observers.
+
+Since TCP eliminates the need for the message layer to support reliability, CoAP over reliable
+transports does not support confirmable or non-confirmable message types. All notifications are
+delivered reliably to the client with positive acknowledgement of receipt occurring at the TCP
+level. If the client does not recognize the token in a notification, it MAY immediately abort
+the connection (see {{sec-abort}}) and SHOULD include a diagnostic payload.
+
+## Cancellation
+
+For CoAP over UDP, a client that is no longer interested in receiving notifications can "forget"
+the observation and respond to the next notification from the server with a reset message to cancel
+the observation. 
+
+For CoAP over reliable transports, a client SHOULD explicitly deregister by issuing a GET request
+that has the Token field set to the token of the observation to be cancelled and includes an Observe
+Option with the value set to 1 (deregister). 
+
+If the client observes one or more resources over a reliable connection, then the CoAP server

```

Comment:
Using the 'endpoint' terminology is possible. Currently, that means referencing the CoAP RFC, which defines the term endpoint quite loosely as: 

4.1.  Messages and Endpoints

   A CoAP endpoint is the source or destination of a CoAP message.  The
   specific definition of an endpoint depends on the transport being
   used for CoAP.  For the transports defined in this specification, the
   **endpoint is identified depending on the security mode used (see
   Section 9): With no security, the endpoint is solely identified by an
   IP address and a UDP port number.  With other security modes, the
   endpoint is identified as defined by the security mode.**

For security modes this endpoint definition does IMHO not really exist. DTLS is the only security option in CoAP and it does not add more information. 

The reason why I spotted this aspect is with regard to devices that sleep a lot. For those, the NAT bindings will expire and consequently the question about demultiplexing an incoming TCP message will arise. My understanding is that an IP address or a port number change will destroy the established state and the client has to re-register from scratch.